sw/device/lib/dif/dif_i2c.h

```diff
@@ -0,0 +1,386 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_I2C_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_I2C_H_
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Represents a speed setting for an I2C component: standard, fast, and
+ * fast plus, corresponding to 100 kbaud, 400 kbaud, and 1 Mbaud,
+ * respectively.
+ *
+ * The integer values of the variants are equal to the maximum speed of
+ * that mode, in kbaud.
+ */
+typedef enum dif_i2c_speed {
+  kDifI2cBusSpeedStandard = 100,
+  kDifI2cBusSpeedFast = 400,
+  kDifI2cBusSpeedFastPlus = 1000,
```

Comment:
Since these values (the values, not the names) aren't used I think making them standard enum values is preferable. If people see that they have value they might think that they can specify values other than the ones present.

Reply:
Fair.